### ISSUE TITLE
[false-positive] “Child method is exactly the same” not checking return type

### ISSUE THREAD

**Issue by rrajkomar** [kalessil/phpinspectionsea#1664]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) , 4.0.6 |
| Language level | PHP 7.4                        |

Given this type of code snippet :
```php
interface ExceptionDefinitionInterface
{
}

interface HttpExceptionDefinitionInterface extends ExceptionDefinitionInterface
{
}

abstract class AbstractExceptionDefinition implements ExceptionDefinitionInterface 
{
    // note that here the methods getCode getValue and getDescription are implemented
}

class HttpExceptionDefinition extends AbstractExceptionDefinition implements HttpExceptionDefinitionInterface
{
}

interface ExceptionInterface
{
    public function getExceptionDefinition(): ExceptionDefinitionInterface;
}

interface HttpExceptionInterface extends ExceptionInterface
{
    public function getHeaders(): array;

    public function getExceptionDefinition(): HttpExceptionDefinitionInterface;
}

abstract class AbstractException extends \RuntimeException implements ExceptionInterface
{
    protected AbstractExceptionDefinition $exceptionDefinition;

    public function __construct(ExceptionInterface $exceptionDefinition, \Exception $previous = null)
    {
        $this->exceptionDefinition = $exceptionDefinition;

        // Some AbstractExceptionDefinition may use a non integer value
        // When that happens, we use 0 as default exception value
        parent::__construct(
            $exceptionDefinition->getDescription(),
            is_int($exceptionDefinition->getValue()) ? $exceptionDefinition->getValue() : 0,
            $previous
        );
    }

    public function getExceptionDefinition(): ExceptionInterface
    {
        return $this->exceptionDefinition;
    }
}

/** @method HttpExceptionDefinition getExceptionDefinition() */
class HttpException extends AbstractException implements HttpExceptionInterface
{
    protected array $headers = [];

    public function __construct(
        HttpExceptionDefinitionInterface $exceptionDefinition,
        array $headers = [],
        \Exception $previous = null
    ) {
        parent::__construct($exceptionDefinition, $previous);
        $this->headers = $headers;
    }

    public function getExceptionDefinition(): HttpExceptionDefinitionInterface
    {
        return $this->exceptionDefinition;
    }

    public function getHeaders(): array
    {
        return $this->headers;
    }
}
```
Current behaviour
--

The method HttpException::getExceptionDefinition gets a "[EA] 'getExceptionDefinition' method can be dropped, as it identical to parent's one." warning because the inspection does not seems to check the return type
(https://github.com/kalessil/phpinspectionsea/blob/master/docs/unused.md#child-method-is-exactly-the-same)

Expected behaviour
--
No warning if return type different.

**Comment by rrajkomar** [comment]
Note : this precise issue can be avoided by not adding the method on the abstract parent and only defining it to the child class.
But it should not thow a warning if it is defined on both levels.

**Comment by ea-inspections-team** [comment]
Thanks for reporting @rrajkomar and providing a great example!

We noticed (see also https://3v4l.org/JcJc2) that '\HttpException::getExceptionDefinition' is not compatible with '\AbstractException::getExceptionDefinition', so there is something bigger happens here (unfortunatelly no upcasting yet in PHP).

In generals for similar cases, following the approach in the constructor (call parent one) is the way to go (еру DRY principle).
